streams/jvm/src/main/scala/zio/stream/experimental/StringDecoder.scala

```diff
@@ -0,0 +1,229 @@
+package zio.stream.experimental
+
+import java.nio.charset.Charset
+import java.nio.charset.StandardCharsets
+import zio.Chunk
+
+object StringDecoder {
+
+  implicit class ByteStreamDecode[R, E, A](val stream: ZStream[R, E, Byte]) extends AnyVal {
+
+    def utfDecode: ZStream[R, E, String]     = new ZStream[R, E, String](stream.channel >>> StringDecoder.utfDecode)
+    def utf8Decode: ZStream[R, E, String]    = new ZStream[R, E, String](stream.channel >>> StringDecoder.utf8Decode)
+    def utf16Decode: ZStream[R, E, String]   = new ZStream[R, E, String](stream.channel >>> StringDecoder.utf16Decode)
+    def utf16BEDecode: ZStream[R, E, String] = new ZStream[R, E, String](stream.channel >>> StringDecoder.utf16BEDecode)
+    def utf16LEDecode: ZStream[R, E, String] = new ZStream[R, E, String](stream.channel >>> StringDecoder.utf16LEDecode)
+    def utf32Decode: ZStream[R, E, String]   = new ZStream[R, E, String](stream.channel >>> StringDecoder.utf32Decode)
+    def utf32BEDecode: ZStream[R, E, String] = new ZStream[R, E, String](stream.channel >>> StringDecoder.utf32BEDecode)
+    def utf32LEDecode: ZStream[R, E, String] = new ZStream[R, E, String](stream.channel >>> StringDecoder.utf32LEDecode)
+    def usASCIIDecode: ZStream[R, E, String] = new ZStream[R, E, String](stream.channel >>> StringDecoder.usASCIIDecode)
+
+  }
+
+  def utfDecode[Err, Done]: ZChannel[Any, Err, Chunk[Byte], Done, Err, Chunk[String], Done] =
+    ZChannel.readOrIdentity { chunk =>
+      if (chunk.startsWith(0 :: 0 :: -2 :: -1 :: Nil) && Charset.isSupported("UTF-32BE"))
```

Comment:
Checking the bytes in this form will box them. You can use `Chunk#byte(Int)` to perform indexed access that won't  box.